File: app/src/main/java/bj/rxjavaexperimentation/model/wantlist/Want.java

package bj.rxjavaexperimentation.model.wantlist;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import bj.rxjavaexperimentation.model.common.RecyclerViewModel;
import bj.rxjavaexperimentation.utils.ArtistsBeautifier;

public class Want implements RecyclerViewModel
{
    @SerializedName("rating")
    @Expose
    private Integer rating;
    @SerializedName("basic_information")
    @Expose
    private BasicInformation basicInformation;
    @SerializedName("resource_url")
    @Expose
    private String resourceUrl;
    @SerializedName("id")
    @Expose
    private Integer id;
    @SerializedName("notes")
    @Expose
    private String notes;
    private String subtitle;

    public Integer getRating()
    {
        return rating;
    }

    public void setRating(Integer rating)
    {
        this.rating = rating;
    }

    public BasicInformation getBasicInformation()
    {
        return basicInformation;
    }

    public void setBasicInformation(BasicInformation basicInformation)
    {
        this.basicInformation = basicInformation;
    }

    public String getResourceUrl()
    {
        return resourceUrl;
    }

    public void setResourceUrl(String resourceUrl)
    {
        this.resourceUrl = resourceUrl;
    }

    public Integer getId()
    {
        return id;
    }

    public void setId(Integer id)
    {
        this.id = id;
    }

    public String getNotes()
    {
        return notes;
    }

    public void setNotes(String notes)
    {
        this.notes = notes;
    }

    @Override
    public String getTitle()
    {
        return basicInformation.getTitle();
    }

    @Override
    public String getSubtitle()
    {
        if (subtitle == null)
        {
            ArtistsBeautifier artistsBeautifier = new ArtistsBeautifier();
            subtitle = artistsBeautifier.formatArtists(basicInformation.getArtists());
        }
        return subtitle;
    }

    @Override
    public String getThumb()
    {
        return basicInformation.getThumb();
    }

    @Override
    public String getType()
    {
        return "release";
    }
}

File: app/src/main/java/bj/discogsbrowser/model/wantlist/Want.kt
package bj.discogsbrowser.model.wantlist

import bj.discogsbrowser.model.common.BasicInformation
import bj.discogsbrowser.model.common.RecyclerViewModel
import bj.discogsbrowser.utils.ArtistsBeautifier
import com.google.gson.annotations.SerializedName

/**
 * Created by Josh Laird on 19/05/2017.
 */
data class Want(var rating: Int = 0,
                @SerializedName("basic_information") var basicInformation: BasicInformation = BasicInformation(),
                @SerializedName("resource_url") var resourceUrl: String = "",
                @get:JvmName("getId_") var id: String = "",
                var notes: String = "",
                @get:JvmName("getSubtitle_") var subtitle: String = ""
) : RecyclerViewModel {

    override fun getTitle(): String {
        return basicInformation.title
    }

    override fun getSubtitle(): String {
        if (subtitle == "") {
            val artistsBeautifier = ArtistsBeautifier()
            subtitle = artistsBeautifier.formatArtists(basicInformation.artists)
        }
        return subtitle
    }

    override fun getThumb(): String {
        return basicInformation.thumb
    }

    override fun getType(): String {
        return "release"
    }

    override fun getId(): String {
        return id
    }
}